src/components/defectReport/DefectReportCreateForm.tsx

```diff
@@ -0,0 +1,309 @@
+import { Alert, Keyboard, StyleSheet } from 'react-native';
+
+import {
+  Button,
+  Checkbox,
+  DropdownInput,
+  HtmlView,
+  ImageSelector,
+  Input,
+  RegularText,
+  Touchable,
+  Wrapper
+} from '../../components';
+import { colors, consts, texts } from '../../config';
+import { CREATE_GENERIC_ITEM } from '../../queries/genericItem';
+import { uploadMediaContent } from '../../queries/mediaContent';
+import { GenericType } from '../../types';
+
+const { EMAIL_REGEX } = consts;
+
+type TDefectReportCreateData = {
+  body: string;
+  categoryName: string;
+  email?: string;
+  images: string;
+  name?: string;
+  phone?: string;
+  termsOfService: boolean;
+  title: string;
+};
+
+export const DefectReportCreateForm = ({
+  navigation,
+  route,
+  selectedPosition,
+  categoryNameDropdownData
+}: {
+  navigation: StackNavigationProp<any>;
+  route: any;
+  selectedPosition: Location.LocationObjectCoords | undefined;
+  categoryNameDropdownData: { id: number; name: string; value: string }[];
+}) => {
+  const consentForDataProcessingText = route?.params?.consentForDataProcessingText ?? '';
+  const [isLoading, setIsLoading] = React.useState(false);
+
+  const {
+    control,
+    formState: { errors },
+    handleSubmit
+  } = useForm({
+    mode: 'onBlur',
+    defaultValues: {
+      body: '',
+      categoryName: '',
+      email: '',
+      images: '[]',
+      name: '',
+      phone: '',
+      termsOfService: false,
+      title: ''
+    }
+  });
+
+  const [createGenericItem, { loading: isLoadingGenericItem }] = useMutation(CREATE_GENERIC_ITEM);
+  let imageUrl: string | undefined;
+
+  const onSubmit = async (defectReportNewData: TDefectReportCreateData) => {
+    Keyboard.dismiss();
+
+    if (!defectReportNewData.termsOfService) {
+      return Alert.alert(texts.defectReport.alerts.hint, texts.defectReport.alerts.termsOfService);
+    }
+
+    setIsLoading(true);
+
+    const images = JSON.parse(defectReportNewData.images);
+
+    if (images?.length) {
+      try {
+        imageUrl = await uploadMediaContent(images[0], 'image');
+      } catch (error) {
+        setIsLoading(false);
+
+        Alert.alert(texts.defectReport.alerts.hint, texts.defectReport.alerts.error);
+        return;
+      }
+    }
+
+    try {
+      await createGenericItem({
+        variables: {
+          categoryName: defectReportNewData.categoryName,
+          genericType: GenericType.DefectReport,
+          title: defectReportNewData.title,
+          contacts: [
+            {
+              email: defectReportNewData.email,
+              firstName: defectReportNewData.name,
+              phone: defectReportNewData.phone
+            }
+          ],
+          contentBlocks: [{ body: defectReportNewData.body, title: defectReportNewData.title }],
+          addresses: [
+            {
+              geoLocation: {
+                latitude: selectedPosition?.latitude,
+                longitude: selectedPosition?.longitude
+              }
```

Comment:
If I press the `Auf der Karte auswählen` button and then press the `weiter` button without selecting a new position on the map, the `selectedPosition` is `undefined`.

<img width="189" alt="image" src="https://user-images.githubusercontent.com/11755668/207925273-2d3063b4-f903-4819-8d5c-f6e11ad65d64.png">

Reply:
very good case, i will have a look and try to set a fallback with the pin we see on the map

Reply:
Very good 👍